src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -227,6 +229,77 @@ module.exports = class LevelPlane extends Array {
       this[myIndex].blockType = "redstoneWireCross";
     }
 
+    if (this[myIndex].isPowered) {
+      this[myIndex].blockType += "On";
+    }
+
     return this[myIndex].blockType;
   }
+
+  getRedstone() {
+    this.redstoneList = [];
+    this.redstoneListON = [];
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType.substring(0, 12) === "redstoneWire") {
+        this[i].isPowered = false;
+        let position = this.indexToCoordinates(i);
+        this.redstoneList.push(position);
+      }
+    }
+    for (let i = 0; i < this.length; ++i) {
```

Comment:
We do this iteration a number of times — let's extract a `getMatchingPositions` helper (or similar) that iterates through the rows and column and takes a callback argument to determine if a given block matches.  The return value would be an array of matching positions.